GetStream/stream-chat-swift
# Fix messages intermittently disappearing when first opening the channel

### 🔗 Issue Links
https://github.com/GetStream/stream-chat-swift/issues/2423
https://github.com/GetStream/ios-issues-tracking/issues/22

### 🎯 Goal
Fixes an issue that messages would disappear when opening a channel for the first time. This is mostly reproducible by changing the `messageLimit = 1` in the `ChannelListQuery`. 

**Bonus:** It also fixes the first page not being loaded from the cache when using `messageLimit = 1`

### 📝 Summary
- Fixes messages intermittently disappearing when first opening the channel
- Fixes the first page not loaded from the cache when using a lower message limit in `ChanneListQuery`

### 🛠 Implementation
It basically reverts #2298, this PR was meant to fix quoted and pinned messages from incorrectly appearing on the first page of the message list, but it caused the 2 issues above.

In order to still fix the original issue with #2298, a fetchLimit and batchSize were added to the CoreData fetch request to make sure the messages are always fetched by batches of the page size. And so, the first page will only fetch 25 messages on the first page, filtering out the pinned or quoted messages that are in the DB. Since the next page will also be 25, and the messages are ordered by `createdAt`, the pinned and quoted messages will always be filtered out, until they actually appear on the page from the network.

### 🧪 Manual Testing Notes
Steps to reproduce the missing messages:
1. Change messageLimit of the Channel List Query to 1 (Right now this can only be done in code, but will soon add a configuration to the DemoApp)
2. Open a Channel
3. Scroll to top and load 1 or 2 more pages
4. Leave the app
5. Open the app
6. Open the same channel
7. Only one message appears
8. Scroll to top
9. One page is skipped, missing messages

### ☑️ Contributor Checklist
